src/taxpy/financial_statements/balance_sheet.py

```diff
@@ -7,64 +7,80 @@
 
 
 class BalanceSheet(FinancialStatement):
-    def add_account(self, name: str, category: str, contra: bool) -> dict[str:dict[str:dict[str:any]]] | None:
-        def create_account(def_bal: str) -> dict[str:dict[str:dict[str:any]]]:
-            """
-            :param def_bal: The default balance of an account (debit/credit).
-            :return: Nothing.
-            """
-            if def_bal == "debit" or def_bal == "credit":
-                self.bal_sht[category][name] = {
-                    "d/c": def_bal,
+    def __init__(self) -> None:
+        """
+        Here is and example of what the balance sheet looks like.\n
+        Type is dict[str:dict[str:dict[str:any]]], if needed.
+
+        self.bs = {
+            "asset": {
+                "cash": {
+                    "d/c": "debit",\n
                     "bal": 0.0
                 }
+            },
 
-                return self.bal_sht[category][name]
-            else:
-                print(f"{def_bal} is not a valid balance type. Please choose either \"debit\" or \"credit\".")
+            "liability": {
+                "accounts payable": {
+                    "d/c": "credit",\n
+                    "bal": 0.0
+                }
+            },
 
+            "equity": {
+                "common stock": {
+                    "d/c": "credit",\n
+                    "bal": 0.0
+                }
+            }
+        }
+        """
+        # Put constructor method at top!
+        self.bal_sht = {
+            "asset": {},
+            "liability": {},
+            "equity": {}
+        }
+        super().__init__()
+    
+    def add_account(self, name: str, category: str, contra: bool) -> dict[str:dict[str:dict[str:any]]]:
+        if category not in ["asset", "liability", "equity"]:
+            raise ValueError("Invalid category type.")
+        
+        def_bal: str = None
         if category == "asset":
-            if contra:
-                return create_account("credit")
-            else:
-                return create_account("debit")
-        elif category == "liability" or category == "equity":
-            if contra:
-                return create_account("debit")
-            else:
-                return create_account("credit")
+            def_bal = "credit" if contra else "debit"
+        # We can assume if not asset then it is either a liability or equity.
```

Comment:
Logically, yes. However, we need to account for the end user making a spelling mistake or putting in a category that isn't `["asset", "liability", "equity"]`. Having a category other than these is not a known practice, so we can't assume it's a liability or equity if it's not an asset.

In the future, support for nonprofit organizations will be added as well, and this will make it easier to implement.